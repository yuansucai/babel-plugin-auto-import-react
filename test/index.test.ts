// @ts-nocheck
import { describe, expect, it } from '@jest/globals';
import autoImportReactPlugin from '../src/index';
import { example1, example2, example3 } from './example';
const babel = require('@babel/core');

const transformOptions = {
  plugins: ['@babel/plugin-syntax-jsx', autoImportReactPlugin],
};

describe('Test auto import react', function () {
    it('Whether the introduction of "React" can be automatically completed, if no react products are introduced', function () {
      const { code } = babel.transform(example1, transformOptions);
      expect(code).toMatchSnapshot();
    });
    it('Whether the introduction of "React" can be automatically completed, if some react products', function () {
      const { code } = babel.transform(example2, transformOptions);
      expect(code).toMatchSnapshot();
    });
    it('If React is already introduced, no changes are made', function () {
      const { code } = babel.transform(example3, transformOptions);
      expect(code).toMatchSnapshot();
    });
});
