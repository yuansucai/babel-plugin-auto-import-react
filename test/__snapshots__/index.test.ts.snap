// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test auto import react If React is already introduced, no changes are made 1`] = `
"import React from 'react';
import _ from 'lodash';
function App() {
  return <div>123</div>;
}"
`;

exports[`Test auto import react Whether the introduction of "React" can be automatically completed, if no react products are introduced 1`] = `
"import React from "react";
function App() {
  return <div>123</div>;
}"
`;

exports[`Test auto import react Whether the introduction of "React" can be automatically completed, if some react products 1`] = `
"import React from "react";
import { useState, useEffect } from 'react';
import { useCallback } from 'react';
function App() {
  const [name, setName] = useState('张三');
  useEffect(() => {
    setName('李四');
  }, []);
  return <div>{name}</div>;
}"
`;
